File: simulator/src/main/java/it/unibo/scafi/simulation/gui/model/Network.java
package it.unibo.scafi.simulation.gui.model;

import java.util.Map;
import java.util.Set;

/**
 * Created by chiara on 14/11/16.
 */
public interface Network {

    Set<Node> getNodes();
    Map<Node, Set<Node>> getNeighborhood();
    Object getPolicy();
    void setNeighborhoodPolicy(Object policy);
    Set<String> getObservableValue();

}

File: simulator/src/main/scala/it/unibo/scafi/simulation/gui/model/Network.scala
package it.unibo.scafi.simulation.gui.model


/**
  * Created by Varini on 14/11/16.
  * Converted/refactored to Scala by Casadei on 3/02/17
  */
trait Network {
  def nodes: Map[Int,Node]

  def neighbourhood: Map[Node, Set[Node]]

  def neighbourhoodPolicy: Any

  def observableValue: Set[String]
}